aws/src/main/python/instanceNotifier/instance_notifier.py

```diff
@@ -0,0 +1,198 @@
+    operating_system = instance.get("PlatformDetails")
+    instance_git_user_email = ""
+    instance_name = ""
+    budget_override = ""
+    tags = instance.get("Tags")
+    for tag in tags:
+        key = tag.get("Key")
+        if key == "GitUserEmail":
+            instance_git_user_email = tag.get("Value")
+        elif key == "Name":
+            instance_name = tag.get("Value")
+        elif key == "BudgetOverride":
+            budget_override = tag.get("Value")
+    return EC2Instance(instance_id, instance_type, operating_system, instance_name, instance_git_user_email, budget_override)
+
+def send_to_sns_notification(message, subject):
+    sns_client.publish(
+        TopicArn='arn:aws:sns:us-east-2:340032650202:instance_notifier',
+        Message=message,
+        Subject=subject)
+
+def stop_instance(instance_id):
+    ec2_client.stop_instances(InstanceIds=[instance_id])
+
+def set_override_tag_to_empty_for_(instance_id):
+    ec2_client.create_tags(
+        Resources=[instance_id],
+        Tags=[
+            {
+                'Key': 'BudgetOverride',
+                'Value': '',
+            },
+        ],
+    )
+
+# Expected format: {"instance-id": <instance-id>, "state": <state>, "region": <region>, "account-id": <account-id>}
+def lambda_handler(event, context):
+    state = event.get('state')
+    if state != "running":
+        return "Instance state change of {state} is not supported for this lambda - only 'running' is considered".format(state=state)
+    region = event.get('region')
+    instance_id = event.get('instance-id')
+    account_id = event.get('account-id')
+
+    init_budget_client()
+    budget_response = budget_client.describe_budget(
+        AccountId=account_id,
+        BudgetName=os.environ["BUDGET_NAME"]
+    )
+
+    budget_details = budget_response.get('Budget')
+    budget_limit_amount = Decimal(budget_details.get('BudgetLimit').get('Amount'))
+    current_spent_amount = Decimal(budget_details.get('CalculatedSpend').get('ActualSpend').get('Amount'))
+    percent_of_budget_used = (current_spent_amount / budget_limit_amount) * 100
+
+    if percent_of_budget_used < 60:
+        return "Budget is not over 60%. Stopping check."
+
+    today = date.today()
+    current_day = Decimal(today.strftime("%d"))
+    current_month = today.strftime("%m")
+    current_year = today.strftime("%Y")
+    days_in_current_month = monthrange(int(current_year), int(current_month))[1]
+    percent_of_month = (current_day / Decimal(days_in_current_month)) * 100
+
+    if percent_of_budget_used <= percent_of_month:
+        return "Budget does not exceed the current point of the month. Stopping check"
+
+    # We are over 60% and over budget AND above percent of month remaining, so notifications must go out
+    # LATER TODO: Add link to budget?
+    # LATER TODO: Send to slack also?
+    # LATER TODO: Add an override option that is not emptied (true_always?) or for # of starts (true_3? - decrement and update tag)
```

Comment:
Should there be a way to deploy an instance with this tag already set? Or would it be better to have this extra step if over budget no matter? In either case a notification would be sent.